lib/iris/experimental/ugrid.py

```diff
@@ -998,97 +1030,614 @@ def equal(self, other, lenient=None):
+            if node_y is not None:
+                self.node_y = node_y
+
+        # deal with the optional members
+        if edge_x is not None:
+            self.edge_x = edge_x
+        if edge_y is not None:
+            self.edge_y = edge_y
+
+    def coord(
+        self,
+        item=None,
+        standard_name=None,
+        long_name=None,
+        var_name=None,
+        attributes=None,
+        axis=None,
+        node=None,
+        edge=None,
+        asdict=True,
+    ):
+        return self._coord(
+            item=item,
+            standard_name=standard_name,
+            long_name=long_name,
+            var_name=var_name,
+            attributes=attributes,
+            axis=axis,
+            node=node,
+            edge=edge,
+            asdict=asdict,
+        )
+
+    def coords(
+        self,
+        item=None,
+        standard_name=None,
+        long_name=None,
+        var_name=None,
+        attributes=None,
+        axis=None,
+        node=None,
+        edge=None,
+        asdict=True,
+    ):
+        # rationalise the tri-state behaviour
+        if node is None and edge is None:
+            node = edge = True
+        else:
+            node = True if node else False
+            edge = True if edge else False
+
+        members = {}
+        if node:
+            members.update(self._filter_members(self.node_coords))
+        if edge:
+            members.update(self._filter_members(self.edge_coords))
+
+        result = self._coords(
+            members,
+            item=item,
+            standard_name=standard_name,
+            long_name=long_name,
+            var_name=var_name,
+            attributes=attributes,
+            axis=axis,
+        )
+
+        if not asdict:
+            result = list(result.values())
+
+        return result
+
+    def remove(
+        self,
+        item=None,
+        standard_name=None,
+        long_name=None,
+        var_name=None,
+        attributes=None,
+        axis=None,
+        node=None,
+        edge=None,
+    ):
+        return self._remove(
+            item=item,
+            standard_name=standard_name,
+            long_name=long_name,
+            var_name=var_name,
+            attributes=attributes,
+            axis=axis,
+            node=node,
+            edge=edge,
+            asdict=True,
+        )
+
+
+class _Mesh2DCoordinateManager(_Mesh1DCoordinateManager):
+    OPTIONAL = (
+        "edge_x",
+        "edge_y",
+        "face_x",
+        "face_y",
+    )
+
+    def __init__(
+        self,
+        node_x,
+        node_y,
+        edge_x=None,
+        edge_y=None,
+        face_x=None,
+        face_y=None,
+    ):
+        super().__init__(node_x, node_y, edge_x=edge_x, edge_y=edge_y)
+
+        # optional coordinates
+        self.face_x = face_x
+        self.face_y = face_y
+
+    def __getstate__(self):
+        # TBD
+        pass
+
+    def __setstate__(self, state):
+        pass
```

Comment:
Are the overrides here necessary? I haven't done this

Reply:
Most likely not. They're only stubs for a reminder to do the work :+1:

Reply:
I've removed them anyways 👍